### ISSUE TITLE
Server crash- onBlockStartBreak

### ISSUE THREAD

**Issue by ProsperCraft** [the-realest-stu/Adventurers-Toolbox#33]
When using a hammer made from a Dragon bone, wither bone, wither bone, amethyst.  The server crashed.

The hammer has the following enchants - Eff 4, Unbreaking III, Veining, Scorching Heat, Fortune III.

When the player resumed mining it did not crash again.

We have Toolbox addons, hence the dragon bone parts - https://minecraft.curseforge.com/projects/toolbox-addons

The Veining enchant comes from the Veining mod- https://minecraft.curseforge.com/projects/veining

Scorching Heat is from Astral Sorcery. https://minecraft.curseforge.com/projects/astral-sorcery

**Comment by ProsperCraft** [comment]
Happened again here - https://gist.github.com/ProsperCraft/690ecf4b04d73d201a398c2bddbc3227

The Scorching heat enchant and the Vening enchant were not on my hammer.  I only have Unbreaking III, Eff V, and Fortune III

**Comment by ProsperCraft** [comment]
Another here- https://gist.github.com/ProsperCraft/a177dc930338478e6fd99d1f81085656

The block I was breaking was - Copper ore from Not Enough Roofs.

https://i.gyazo.com/2a436d5a5d981bf8c1d03670104a8288.png

**Comment by ProsperCraft** [comment]
Happened again-
https://gist.github.com/ProsperCraft/75ba171c41827062e0707116f3d10d63

Here is what the console shows when it happens - 
https://gist.github.com/ProsperCraft/a7646dafe287bb53cb18f001b192d706

**Comment by WolfieWaffle** [comment]
Can you try to find the crash with only vanilla and Toolbox? Just to make sure it's this mod. It might be fixable from here but I want to find out where the problem is specifically, like what mod. I am assuming it is something with Vein Miner since that would relate to breaking blocks.

**Comment by ProsperCraft** [comment]
Yes, I suspected the vein miner enchant, until I had it happen with a hammer that did not have it on it.  

Can you not tell from the stacktrace?

**Comment by ProsperCraft** [comment]
I mined for hours and hours with no crashes, then when another play logged in and was not mining, I crashed within minutes, think it is related? 

https://gist.github.com/ProsperCraft/9aab2ec5f1675b52421a394cdfbf9409

**Comment by the-realest-stu** [comment]
This may have been fixed in version 0.2.12, which was just submitted to curseforge. Please let me know if the issue persists after upgrading to 0.2.12.
